### PR TITLE
Fixes to S3 catalog.

### PR DESCRIPTION
**Pull request by lossyrob**
S3 Catalog had some issues:
 - reader was not truncating/padding indexes like writer was
 - some indexes were being skipped over by readKeys logic
 - configuration was not being passed to the AmazonS3Client.

### CODE REVIEW

File: spark/src/main/scala/geotrellis/spark/io/s3/RasterRDDReader.scala [hunk 1]
```diff
@@ -38,36 +40,41 @@ abstract class RasterRDDReader[K: ClassTag] extends LazyLogging {
...
-            listKeys(s3client, bucket, dir, range, itp, pti)            
-              .map{ path => 
-                val is = s3client.getObject(bucket, path).getObjectContent
-                KryoSerializer.deserializeStream[(K, Tile)](is)
-              }
-              .filter{ row => filterSet.includeKey(row._1) }
-          }
-      }
+      sc
+        .parallelize(bins, bins.size)
```
Comment (notthatbreezy) [inline spark/src/main/scala/geotrellis/spark/io/s3/RasterRDDReader.scala]: I wonder if this is a place where we could take advantage of the built-in s3 and binary records support for spark -- maybe something like

```scala
sc.binaryRecords("s3a://file1,s3a://file2...", numPartitions)
                            .map{ case (filename, bytes) => KryoSerializer.deserializeStream[(K, Tile)](bytes) }
                            .filter{ row => filterSet.includeKey(row._1) }
```

Comment (notthatbreezy) [inline spark/src/main/scala/geotrellis/spark/io/s3/RasterRDDReader.scala]: Actually, that would probably `binaryFiles` I think -- would prevent from worrying about binning etc, deferring that logic to spark as long as we know the list of keys

Comment (echeipesh) [inline spark/src/main/scala/geotrellis/spark/io/s3/RasterRDDReader.scala]: Would be interesting to try, you would have to list all the files first on the master. I'm not sure if that would have a significant impact.

**Comment by echeipesh** [comment]
+1

File: spark/src/main/scala/geotrellis/spark/io/s3/RasterRDDReader.scala (hunk 1 above)
Comment (notthatbreezy) [inline spark/src/main/scala/geotrellis/spark/io/s3/RasterRDDReader.scala]: Oh that's a good point

**Pull request merged**